Assets/Scripts/Framework/Gameplay/LevelTracker.cs

```diff
@@ -0,0 +1,208 @@
+            }
+
+            return labels;
+        }
+
+        private List<Transform> SpawnDeliveries(StationLevelData[] levelData, List<string> labels)
+        {
+            List<Transform> usedSpawns = new();
+
+            for (int i = levelData.Length - 1; i >= 0; i--)
+            {
+                List<Transform> availableSpawns = GetAvailableSpawnsForStation(i, usedSpawns);
+                int amountToSpawn = levelData[i].minDeliverablePackages;
+
+                while (amountToSpawn > 0 && availableSpawns.Count > 0)
+                {
+                    availableSpawns.Shuffle();
+                    Transform targetSpawn = availableSpawns[0];
+
+                    if (usedSpawns.Contains(targetSpawn))
+                        continue;
+
+                    PackageData package = SpawnBoxDelivery(targetSpawn, labels[i], i);
+
+                    amountToSpawn--;
+                    usedSpawns.Add(targetSpawn);
+                    availableSpawns.Remove(targetSpawn);
+
+                    _totalScore += _succesfullDeliveryBonus + package.GetComponent<BoxDamageable>().Health;
+                }
+            }
+
+            return usedSpawns;
+        }
+
+        private void SpawnFakeDeliveries(List<Transform> spawns, List<string> usedLabels)
+        {
+            System.Random random = new System.Random();
+            spawns.Shuffle();
+            int fakeDeliverySpawns = random.Next(0, spawns.Count);
+
+            for (int i = 0; i < fakeDeliverySpawns; i++)
+            {
+                string label = string.Empty;
+                while (label == string.Empty || usedLabels.Contains(label))
+                {
+                    label
+                        = CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString()
+                        + CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString();
+                }
+
+                SpawnBoxDelivery(spawns[i], label, -1);
+            }
+        }
+
+        private List<Transform> GetAvailableSpawnsForStation(int stationIndex, List<Transform> usedSpawns)
+        {
+            List<Transform> spawns = new();
+
+            for (int i = stationIndex - 1; i >= 0; i--)
+                for (int j = 0; j < _allStations[i].SpawnTransforms.Length; j++)
+                    if (!usedSpawns.Contains(_allStations[i].SpawnTransforms[j]))
+                        spawns.Add(_allStations[i].SpawnTransforms[j]);
+
+            return spawns;
+        }
+
+        private List<Transform> GetAllFreeSpawns(List<Transform> usedSpawns)
+        {
+            List<Transform> freeSpawns = new();
+
+            for (int i = 0; i < _allStations.Length; i++)
+                for (int j = 0; j < _allStations[i].SpawnTransforms.Length; j++)
+                    if (!usedSpawns.Contains(_allStations[i].SpawnTransforms[j]))
+                        freeSpawns.Add(_allStations[i].SpawnTransforms[j]);
+
+            return freeSpawns;
+        }
```

Comment:
This can be made more clear using LINQ. It would look like this:
```suggestion
        private List<Transform> GetAllFreeSpawns(List<Transform> usedSpawns)
        {
            return _allStations
                .SelectMany(station => station.SpawnTransforms)
                .Except(usedSpawns)
                .ToList();
        }
```